Classes/Emogrifier.php

```diff
@@ -323,6 +334,77 @@ protected function process(\DOMDocument $xmlDocument)
     }
 
     /**
+     * Applies $newStyles to $node.
+     *
+     * This method maps CSS styles to HTML attributes and adds those to the
+     * node.
+     *
+     * @param string[] $newStyles the new CSS styles taken from the global styles
+     *                            to be applied to this node.
+     * @param \DOMNode $node      node to apply styles to.
+     *
+     * @return void
+     */
+    private function mapStyles2HtmlAttrs(array $newStyles, \DOMNode $node)
+    {
+        foreach ($newStyles as $styleName => $styleValue) {
```

Comment:
As a first step to simplify this method, I propose moving the coding within the loop to a separate, well-name method.